libs/libGamedata/lua/CampaignDataLoader.cpp

```diff
@@ -28,8 +28,10 @@ bool CampaignDataLoader::Load()
     auto curFile_ = basePath_ / "campaign.lua";
     try
     {
-        if(!loadScript(curFile_))
+        if(!loadScript(curFile_) || lua["campaign"].isNilref() || lua["campaign"].type() != LUA_TTABLE)
             return false;
+
+        campaignDesc_ = initFromLuaTable(lua["campaign"]);
```

Comment:
With the above suggestion:

```suggestion
        campaignDesc_ = CampaignDescription(lua["campaign"]);
```

Reply:
Done